File: Ruby/c4.rb
require_relative './connect4'

rows = 6
cols = 7
win_length = 4

if ARGV.length >= 1
    if ARGV[0].strip =~ /^(\d+)x(\d+)$/
        rows = $1.to_i
        cols = $2.to_i
    else
        puts "Board size \"#{ARGV[0]}\" is not formatted properly."
        exit
    end
end

if ARGV.length >= 2
    if ARGV[1].strip =~ /^(\d+)$/
        win_length = $1.to_i
    else
        puts "Win length must be an integer"
        exit
    end
end

c4 = Connect4.new(rows, cols, win_length)
c4.play_game

File: JavaScript/c4.mjs
import { exit } from "process";       
import Connect4 from "./Connect4.mjs"  

const myArgs = process.argv.slice(2);

let rows = 6;
let cols = 7;
let winLength = 4;


if (myArgs.length >= 1) {
    let matches = /(\d+)x(\d+)/.exec(myArgs[0]);
    if (matches === null) {
        console.log(`Board size "${myArgs[0]}" is not formatted properly.`);
        exit(); 
    } else {
        // matches are strings
        rows = parseInt(matches[1]);
        cols = parseInt(matches[2]);
        winLength = process.argv[3];
    }
}

(new Connect4(rows, cols, winLength)).playGame(1, 0, () => {});
